src/adapter/http/src/http_server_dataset_router.rs

```diff
@@ -150,14 +158,294 @@ pub async fn platform_login_handler(
         ("api_key" = []),
     )
 )]
-#[allow(clippy::unused_async)]
+#[expect(clippy::unused_async)]
 pub async fn platform_token_validate_handler(catalog: Extension<Catalog>) -> Result<(), ApiError> {
     ensure_authenticated_account(&catalog).api_err()?;
     Ok(())
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+/// <https://datatracker.ietf.org/doc/html/rfc8628#section-3.1>
```

Comment:
This file became very big. Consider splitting on multiple Rust files.

Reply:
Extracted all platform related to `./platform` directory

Done